src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -150,29 +167,42 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
         logger.trace("created new InternalEngine");
     }
 
-    private SearcherManager createSearcherManager() throws EngineException {
+
+    @Override
+    public TranslogRecoveryStats getInitialTranslogRecoveryStats() {
+        return this.translogRecovery;
+    }
+
+    /**
+     * Reads the current stored translog ID (v1) from the IW commit data and generates a new/next translog ID (v2)
+     * from the largest present translog ID. If there is no stored translog ID both IDs are identical.
+     */
+    private Tuple<Long, Long> loadTranslogIds(IndexWriter writer, Translog translog) throws IOException {
+        // commit on a just opened writer will commit even if there are no changes done to it
+        // we rely on that for the commit data translog id key
+        final long nextTranslogId = Math.max(0, translog.findLargestPresentTranslogId()) + 1;
+        final Map<String, String> commitUserData = writer.getCommitData();
+        if (commitUserData.containsKey(Translog.TRANSLOG_ID_KEY)) {
+            final long currentTranslogId = Long.parseLong(commitUserData.get(Translog.TRANSLOG_ID_KEY));
+            return new Tuple<>(currentTranslogId, nextTranslogId);
+        }
+         // translog id is not in the metadata - fix this inconsistency some code relies on this and old indices might not have it.
+        writer.setCommitData(Collections.singletonMap(Translog.TRANSLOG_ID_KEY, Long.toString(nextTranslogId)));
```

Comment:
can we add an info, if not warn message here? I wonder how old the index needs to be  for this to kick in. Do we know?